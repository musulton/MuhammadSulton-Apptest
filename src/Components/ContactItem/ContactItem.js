import * as React from 'react';
import {Image, View, Text, TouchableOpacity} from 'react-native';

import styles from './ContactItem.styles';

const Bio = ({firstName, lastName}) => (
  <View style={styles.bioContainer}>
    <Text style={styles.fullName}>
      {firstName} {lastName}
    </Text>
  </View>
);

const Photo = ({photo}) => <Image source={{uri: photo}} style={styles.photo} />;

const ContactItem = ({data, onPress}) => (
  <TouchableOpacity style={styles.container} onPress={onPress}>
    <Photo photo={data.photo} />
    <Bio firstName={data.firstName} lastName={data.lastName} />
  </TouchableOpacity>
);

export default ContactItem;
