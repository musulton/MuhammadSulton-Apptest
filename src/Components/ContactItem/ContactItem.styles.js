import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    height: 75,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  photo: {
    width: 50,
    height: 50,
    borderRadius: 50,
    marginRight: 10,
  },
  bioContainer: {
    flex: 1,
  },
  fullName: {
    fontSize: 20,
    fontWeight: '400',
  },
  ageContainer: {
    width: 40,
    padding: 2,
    backgroundColor: 'green',
    borderRadius: 20,
    marginTop: 5,
    alignItems: 'center',
  },
  age: {
    fontSize: 14,
    color: 'white',
  },
});
