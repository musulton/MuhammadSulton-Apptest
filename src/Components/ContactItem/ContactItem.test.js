import * as React from 'react';
import {render, cleanup} from '@testing-library/react-native';

import ContactItem from './ContactItem';

jest.mock('../index', () => ({
  Avatar: 'Avatar',
}));

describe('ContactItem', () => {
  afterEach(cleanup);

  const data = {
    firstName: 'firstName',
    lastName: 'lastName',
    age: 20,
    photo:
      'https://static.wikia.nocookie.net/naruto/images/d/dd/Naruto_Uzumaki%21%21.png/revision/latest?cb=20161013233552',
  };

  it('should render contact item correctly', () => {
    const {toJSON} = render(<ContactItem data={data} onPress={jest.fn} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
