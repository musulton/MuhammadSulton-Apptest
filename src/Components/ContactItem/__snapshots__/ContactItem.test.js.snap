// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactItem should render contact item correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 75,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Avatar
    size={60}
    uri="https://static.wikia.nocookie.net/naruto/images/d/dd/Naruto_Uzumaki%21%21.png/revision/latest?cb=20161013233552"
  />
  <View
    style={
      Object {
        "marginHorizontal": 6,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 20,
          "fontWeight": "400",
        }
      }
    >
      firstName
       
      lastName
    </Text>
  </View>
</View>
`;
